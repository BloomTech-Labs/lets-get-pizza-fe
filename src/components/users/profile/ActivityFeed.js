import React from 'react'
import { Feed } from 'semantic-ui-react'
import EventActivity from './EventActivity'
import ReviewActivity from './ReviewActivity'

const ActivityFeed = ({ activities, user }) => {
    return (
        <Feed size="large" style={{width: '80%', margin: '0 auto'}}>
            {activities.map((activity, index) => (
                activity.start_time ? 
                <EventActivity key={index} user={user} event={activity}/> :
                <ReviewActivity key={index} user={user} review={activity} />
            ))}
        </Feed>
    )
}

export default ActivityFeed