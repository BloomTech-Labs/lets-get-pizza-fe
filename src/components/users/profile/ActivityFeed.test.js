import React from "react";
// import {
//   render,
//   fireEvent,
//   cleanup,
//   waitFor,
//   waitForElementToBeRemoved,
// } from "@testing-library/react";
// import "@testing-library/jest-dom/extend-expect";
// import { findFirstProp } from "../../../utils/reduxTestingFunctions";
import renderer from "react-test-renderer";

import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import ActivityFeed from "./ActivityFeed";

const mockStore = configureStore({});

describe("activities render", () => {
  let store;
  let component;

  const activities = [
    {
      address: "34579 Yucaipa Blvd (5th St), Yucaipa, CA 92399, United States",
      business_name: "Pizza Chalet",
      id: 14,
      location_id: 33,
      rating: 5,
      review_text: "I used to go here all the time as a child",
      review_title: "Nostalgia",
      user_id: 10,
    },
  ];
  const user = {
    username: "AnimeCody",
  };

  beforeEach(() => {
    store = mockStore({
      user: {
        username: "AnimeCody",
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <ActivityFeed activities={activities} user={user} />
      </Provider>
    );
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });
});
