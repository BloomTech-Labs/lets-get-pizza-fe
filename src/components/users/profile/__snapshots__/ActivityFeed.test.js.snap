// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activities render renders 1`] = `
<div
  className="ui large feed"
  style={
    Object {
      "margin": "0 auto",
      "width": "80%",
    }
  }
>
  <div
    className="event"
  >
    <div
      className="label"
    />
    <div
      className="content"
    >
      <div
        className="summary"
      >
        <a
          className="user"
          href="/users/dash/profile"
        >
          AnimeCody
        </a>
         
        posted a review of 
        Pizza Chalet
      </div>
      <br />
      <div
        className="summary"
      >
        Nostalgia
         
        <div
          className="ui disabled rating"
          onMouseLeave={[Function]}
          role="radiogroup"
          tabIndex={0}
        >
          <i
            aria-checked={false}
            aria-posinset={1}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={-1}
          />
          <i
            aria-checked={false}
            aria-posinset={2}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={-1}
          />
          <i
            aria-checked={false}
            aria-posinset={3}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={-1}
          />
          <i
            aria-checked={false}
            aria-posinset={4}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={-1}
          />
          <i
            aria-checked={true}
            aria-posinset={5}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={-1}
          />
        </div>
      </div>
      <div
        className="text extra"
      >
        I used to go here all the time as a child
      </div>
      <div
        className="meta"
      >
        <a
          className="like"
        >
          <i
            aria-hidden="true"
            className="thumbs up icon"
            onClick={[Function]}
          />
        </a>
         13 Likes
      </div>
       
    </div>
  </div>
  <div
    className="ui divider"
  />
</div>
`;
